### PR TITLE
[JBPM-10059] Aborted staged remains active in the process engine

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-10059](https://issues.redhat.com/browse/JBPM-10059)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
I think we need to test for JBPM-10059, For my side, the case our node status is not canceled.

**Comment by gmunozfe** [comment]
> I think we need to test for JBPM-10059, For my side, In the case the node status is not canceled.

I've added this test: `testCancelStageForMultiInstanceStages` where 2 stages are instantiated (active) and other remains available. Then, the test cancels one stage and gets stages (both, `onlyActive` set to false and true) and then, cancels the other one. I will add one more to cover also "completed" stages, thanks for your comment @bxf12315 !

**Comment by gmunozfe** [comment]
> > I think we need to test for JBPM-10059, For my side, In the case the node status is not canceled.
> 
> I've added this test: `testCancelStageForMultiInstanceStages` where 2 stages are instantiated (active) and other remains available. Then, the test cancels one stage and gets stages (both, `onlyActive` set to false and true) and then, cancels the other one. I will add one more to cover also "completed" stages, thanks for your comment @bxf12315 !

Done, now the test is working with all stage statuses (Available, Active, Canceled and Completed).

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by bxf12315** [comment]
Jenkins run fdb

**Review by elguardian — APPROVED** [review]
LGTM

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

File: jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCancelTest.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+package org.jbpm.casemgmt.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
```
Comment (afalhambra) [inline jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCancelTest.java:25]: Not used, can be removed

File: jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCancelTest.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Completed))
+            .noneMatch(item -> item.getStatus().equals(StageStatus.Active))
+            .filteredOn(item -> item.getStatus().equals(StageStatus.Canceled)).hasSize(2);
+            
+            assertActiveStages(caseId, 0);
+
+        } catch (Exception e) {
+            fail("Unexpected exception " + e.getMessage());
+        } finally {
+            if (caseId != null) {
```
Comment (afalhambra) [inline jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCancelTest.java:147]: this check is not needed, as caseId will never be equal to null at this point due to `assertNotNull(caseId);` at the beginning

**Review by afalhambra — APPROVED** [review]
Outstanding work @gmunozfe! nailed down

**Pull request merged**